pkg/controller/certificates/trigger/trigger_controller.go

```diff
@@ -147,9 +147,25 @@ func (c *controller) ProcessItem(ctx context.Context, key string) error {
 		return nil
 	}
 
-	// ensure a resync is scheduled in the future so that we re-check
-	// Certificate resources and trigger them near expiry time
-	c.scheduleRecheckOfCertificateIfRequired(log, key, crt.Status.RenewalTime)
+	// check if we have had a recent failure, and if so do not trigger a
+	// reissue immediately
+	if crt.Status.LastFailureTime != nil {
+		// the amount of time after the LastFailureTime before the request should be retried
+		const retryAfterLastFailure = time.Hour
+		now := c.clock.Now()
+		retryAfter := crt.Status.LastFailureTime.Add(retryAfterLastFailure)
+		if now.Before(retryAfter) {
+			log.Info("Not re-issuing certificate as an attempt has been made in the last hour", "retry_after", retryAfter)
+			c.scheduleRecheckOfCertificateIfRequired(log, key, retryAfter)
+			return nil
+		}
+	}
+
+	if crt.Status.RenewalTime != nil {
+		// ensure a resync is scheduled in the future so that we re-check
+		// Certificate resources and trigger them near expiry time
+		c.scheduleRecheckOfCertificateIfRequired(log, key, crt.Status.RenewalTime.Time)
```

Comment:
Could it be that a recheck is scheduled twice? If failed in the last hour and `Status.Renewaltime` is not nil?

Reply:
No, because the `LastFailureTime != nil` case, if it does schedule a renewal, will `return` immediately after.

Additionally, the scheduler package 'de-duplicates' items already, so there'll only ever be one instance scheduled even if we _did_ mess this area of code up 😄